lib/js/jsfetch.nim

```diff
@@ -0,0 +1,78 @@
+## - Fetch for the JavaScript target: https://developer.mozilla.org/docs/Web/API/Fetch_API
+from asyncjs import PromiseJs
+from httpcore import HttpMethod
+export PromiseJs, HttpMethod
+
+when not defined(js) and not defined(nimdoc):
+  {.fatal: "Module jsfetch is designed to be used with the JavaScript backend.".}
+when defined(nodejs):
+  {.warning: "By design 'fetch()' is defined on the web browser, but may not be on NodeJS.".}
+
+template method2cstring(metod: HttpMethod): cstring =
+  ## Template that takes an `HttpMethod` and returns an *Uppercase* `cstring`,
+  ## but *only* for the HTTP Methods that are supported by the fetch API.
+  ## High performance and minimal code compared to just `$(HttpMethod)`.
+  assert metod notin {HttpTrace, HttpOptions, HttpConnect}, "HTTP Method not supported by fetch API"
+  case metod
+  of HttpHead:   "HEAD".cstring
+  of HttpGet:    "GET".cstring
+  of HttpPost:   "POST".cstring
+  of HttpPut:    "PUT".cstring
+  of HttpDelete: "DELETE".cstring
+  of HttpPatch:  "PATCH".cstring
+  else:          "GET".cstring
+
+func fetch*(url: cstring): PromiseJs {.importcpp: "fetch(#)".}
+  ## `fetch()` API, Simple GET only (generates `fetch(url)`).
+  ##
+  ## .. code-block:: nim
+  ##   doAssert fetch("https://nim-lang.org") is PromiseJs
+
+func fetch*(url: cstring, httpMethod: static[HttpMethod], body: cstring): PromiseJs {.asmNoStackFrame.} =
+  ## `fetch()` API, for any HTTP Method (static overload for optimization).
+  ##
+  ## .. code-block:: nim
+  ##   doAssert fetch("https://my.api.org", HttpPost, "body") is PromiseJs
+  # See the generated code if you dont understand why this overload exists.
+  assert url.len > 0, "url must not be empty string"
+  const x: cstring = method2cstring(httpMethod)
+  {.emit: """return fetch(`url`, {method: "`x`", body: `body`});""".}
+
+func fetch*(url: cstring, httpMethod: HttpMethod, body: cstring): PromiseJs {.asmNoStackFrame.} =
+  ## .. code-block:: nim
+  ##   let mthd = HttpPost
```

Comment:
argument only used once, inline it directly `doAssert fetch("https://my.api.org", HttpPost, "body") ...` (but same comment as above)

Reply:
No, different overload.

```nim
func fetch*(url: cstring, httpMethod: static[HttpMethod], body: cstring): PromiseJs {.asmNoStackFrame.} =
  const x: cstring = method2cstring(httpMethod)
  {.emit: """return fetch(`url`, {method: "`x`", body: `body`});""".}
```
:arrow_down: :arrow_down: :arrow_down: :arrow_down: 
```javascript
function fetch_10201036(url_10201039, body_10201041) {
    return fetch(url_10201039, {method: "POST", body: body_10201041});
}
fetch_10201036("https://my.api.org", "body");
```

---

```nim
func fetch*(url: cstring, httpMethod: static[HttpMethod], body: cstring): PromiseJs {.asmNoStackFrame.} =
  let x: cstring = method2cstring(httpMethod)
  {.emit: """return fetch(`url`, {method: "`x`", body: `body`});""".}
```
:arrow_down: :arrow_down: :arrow_down: :arrow_down: 
```javascript
function fetch_10201036(url_10201039, body_10201041) {
    var Tmp1;
    switch (2) {
    case 0:
      Tmp1 = "HEAD";
      break;
    case 1:
      Tmp1 = "GET";
      break;
    case 2:
      Tmp1 = "POST";
      break;
    case 3:
      Tmp1 = "PUT";
      break;
    case 4:
      Tmp1 = "DELETE";
      break;
    case 8:
      Tmp1 = "PATCH";
      break;
    default: 
      Tmp1 = "GET";
      break;
    }
    var x_10220201 = Tmp1;
    return fetch(url_10201039, {method: x_10220201, body: body_10201041});
}
fetch_10201036("https://my.api.org", "body");
```

Reply:
why do we need a static overload? I see no practical performance benefit